### ISSUE TITLE
DHF Hardening: empty collector result in failed job.

### ISSUE THREAD

**Issue by josvanroosmalen** [marklogic/marklogic-data-hub#1715]
If you update a collector to:

```
function collect(options) {
   return []
}
```

And run the harmonization, this will result in:

```
"status": "FAILED", 
"successfulEvents": 0, 
"failedEvents": 0, 
"successfulBatches": 0, 
"failedBatches": 0
```
The status should be **FINISHED**. All data of collector (which is empty) is processed. 

**The current approach makes it difficult to use DHF in a Daemon situation** (eg. where you run this every x interval, where situation can happen that the collector return 0 results). In that case you cannot rely on the job status.

**Comment by akshaysonvane** [comment]
Thanks for filing this issue.
Tracking it via https://project.marklogic.com/jira/browse/DHFPROD-1668

**Comment by sbayatpur** [comment]
Closing this issue – it was fixed in 4.2.2.
